### PR TITLE
nto pkg: add WithAnnotation and WithNet methods to the performanceprofile

### PR DESCRIPTION
**Pull request by elenagerman**

### CODE REVIEW

File: pkg/nto/performanceprofile_test.go [hunk 1]
```diff
@@ -41,6 +41,17 @@ var (
 			Node:  &defaultHugePagesNodeTwo,
 		},
 	}
+	defaultAnnotationOne = map[string]string{"performance.openshift.io/ignore-cgroups-version": "true"}
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:44]: This var use only once. Just copy paste map[string]string{"performance.openshift.io/ignore-cgroups-version": "true"} directly to the testCases.testAnnotations

File: pkg/nto/performanceprofile_test.go [hunk 2]
```diff
@@ -41,6 +41,17 @@ var (
...
+	defaultAnnotationTwo = map[string]string{
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:45]: Same as above

File: pkg/nto/performanceprofile_test.go [hunk 3]
```diff
@@ -41,6 +41,17 @@ var (
...
+		"kubeletconfig.experimental": "{'systemReserved':{'cpu':'500m','memory':'28Gi'}}",
+	}
+	defaultAnnotationThree = map[string]string{"performance.openshift.io/ignore-cgroups-version": "true",
```
Comment (kononovn) [inline pkg/nto/performanceprofile_test.go:48]: Same as above

**Review by kononovn — CHANGES_REQUESTED** [review]
Looks good just few cosmetic issues.

File: pkg/nto/performanceprofile_test.go [hunk 4]
```diff
@@ -604,6 +609,145 @@ func TestPerformanceProfileWithWorkloadHints(t *testing.T) {
...
+			assert.NotNil(t, result)
+			assert.Equal(t, testCase.testAnnotations, result.Definition.ObjectMeta.Annotations)
+		}
+	}
+}
+
+//nolint:funlen
+func TestPerformanceProfileWithNet(t *testing.T) {
+	testCases := []struct {
+		testUserLevelNet  bool
```
Comment (achuzhoy) [inline pkg/nto/performanceprofile_test.go:655]: Do we want any test with testUserLevelNet set to false? All the tests are true.

Comment (elenagerman) [reply pkg/nto/performanceprofile_test.go:655]: ups, one sec adding

**Review by cdvultur — APPROVED** [review]
/lgtm

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**